.NET8 AOT produces faulty win-x64 static library

### Description

Win-x64 static library produced by .NET8 AOT fails to link when used in a Windows c++ console project. It results in error by cvtres.exe: ConsoleApplication1.res : fatal error LNK1241: resource file ClassLibrary1.lib(ClassLibrary1.obj) already specified.

Removing .rc file from the build solves the issue.

### Reproduction Steps

1. Create .NET8 C# class library project. Add <PublishAot>true</PublishAot><NativeLib>Static</NativeLib> to the project file. Publish as win-x64.
2. Create a C++ desktop console project.  Switch to static c-runtime. Link against the published library, Bcrypt.lib and (...\.nuget\packages\runtime.win-x64.microsoft.dotnet.ilcompiler\8.0.0\sdk\...)
    bootstrapperdll.obj
    Runtime.WorkstationGC.lib
    eventpipe-enabled.lib
    Runtime.VxsortEnabled.lib
    System.Globalization.Native.Aot.lib
    System.IO.Compression.Native.Aot.lib
3. Build the app.


### Expected behavior

Builds w/o issues

### Actual behavior

ConsoleApplication1.res : fatal error LNK1241: resource file ClassLibrary1.lib(ClassLibrary1.obj) already specified.

### Regression?

Was linking successfully in .NET7

### Known Workarounds

Remove .rc from the build

### Configuration

_No response_

### Other information

_No response_

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
@@ -262,7 +262,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <IlcArg Condition="$(IlcMaxVectorTBitWidth) != ''" Include="--max-vectort-bitwidth:$(IlcMaxVectorTBitWidth)" />
       <IlcArg Condition="$(IlcSingleThreaded) == 'true'" Include="--parallelism:1" />
       <IlcArg Condition="$(IlcSystemModule) != ''" Include="--systemmodule:$(IlcSystemModule)" />
-      <IlcArg Condition="'$(_targetOS)' == 'win' and $(IlcMultiModule) != 'true' and '$(IlcGenerateWin32Resources)' != 'false'" Include="--win32resourcemodule:%(ManagedBinary.Filename)" />
+      <IlcArg Condition="'$(_targetOS)' == 'win' and $(IlcMultiModule) != 'true' and '$(IlcGenerateWin32Resources)' != 'false' and '$(NativeLib)' != 'Static'" Include="--win32resourcemodule:%(ManagedBinary.Filename)" />
       <IlcArg Condition="$(IlcDumpIL) == 'true'" Include="--ildump:$(NativeIntermediateOutputPath)%(ManagedBinary.Filename).il" />
       <IlcArg Condition="$(NoWarn) != ''" Include='--nowarn:"$([MSBuild]::Escape($(NoWarn)).Replace(`%0A`, ``).Replace(`%0D`, ``))"' />
       <IlcArg Condition="$(TrimmerSingleWarn) == 'true'" Include="--singlewarn" />

